src/librustc_resolve/lib.rs

```diff
@@ -1989,7 +1989,31 @@ impl<'a> Resolver<'a> {
             // When resolving `$crate` from a `macro_rules!` invoked in a `macro`,
             // we don't want to pretend that the `macro_rules!` definition is in the `macro`
             // as described in `SyntaxContext::apply_mark`, so we ignore prepended modern marks.
-            ctxt.marks().into_iter().rev().find(|m| m.transparency() != Transparency::Transparent)
+            // FIXME: This is only a guess and it doesn't work correctly for `macro_rules!`
```

Comment:
Can we just forbid `macro` to produce `macro_rules` and `macro_rules` to produce `macro`? Or is that something we'll want to support

Reply:
I still hope to fix this by replacing the context transplantation trick (see below) with some less ambiguous representation.